mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -172,37 +274,78 @@ object StringIndexer extends DefaultParamsReadable[StringIndexer] {
 
   @Since("1.6.0")
   override def load(path: String): StringIndexer = super.load(path)
+
+  // Returns a function used to sort strings by frequency (ascending or descending).
+  // In case of equal frequency, it sorts strings by alphabet (ascending).
+  private[feature] def getSortFunc(
+      ascending: Boolean): ((String, Long), (String, Long)) => Boolean = {
+    if (ascending) {
+      { case ((strA: String, freqA: Long), (strB: String, freqB: Long)) =>
```

Comment:
NIt: the indentation is still off below. I think that you can omit the outer braces and case, as this isn't a partial function:

```
if (ascending) {
  ((strA, ...), (..., ...)) =>
    if (...) {
```

Reply:
Looks the outer braces can be omitted, but the `case` can't.